src/RomaricDrigon/MetaYaml/Exception/NodeValidatorException.php

```diff
@@ -5,16 +5,50 @@
 class NodeValidatorException extends \Exception
 {
     private $node_path;
+    private $paths = [];
+    private $messages = [];
 
-    public function __construct($node_path, $message)
+  public function __construct($node_path, $messages, $paths)
     {
         $this->node_path = $node_path;
+        if (!is_array($messages)) {
+          $messages = [$messages];
+        }
+        if (!is_array($paths)) {
+          $paths = [$paths];
+        }
+        $this->paths = $paths;
+        $this->messages = $messages;
 
-        parent::__construct($message);
+        parent::__construct(current($messages));
     }
 
     public function getNodePath()
     {
         return $this->node_path;
     }
+
+    public function getPaths()
+    {
+        return $this->paths;
+    }
+
+    public function getMessages()
+    {
+        return $this->messages;
+    }
+
+    public function getReport()
```

Comment:
i'm not sure to understand what `getReport()`is giving? As it will become part of MetaYaml API, there should be a test over it.